torch/csrc/jit/frontend/lexer.h

```diff
@@ -241,12 +241,12 @@ struct TORCH_API SharedParserData {
     // invariant: the next token is not whitespace or newline
     *start = pos;
     // check for a valid number
-    if (isNumber(str, pos, len)) {
+    if (isNumber(str.piece(0), pos, len)) {
```

Comment:
Maybe `StringCordView` should have a method that's called `only_piece`, which returns `piece(0)` and asserts there is only a single piece?

Reply:
That won't work because if I save a Source then load it again, then its StringCordView is no longer only_piece.
This change is correct if in StringCordView never split a token into 2 pieces.